### PR TITLE
Add jacobians for `LieGroup.local()`

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: theseus/geometry/lie_group.py [hunk 1]
```diff
@@ -167,13 +167,22 @@ def _compose_jacobian(
...
+        diff = self.between(variable2)
 
-    def _local_jacobian(self, variable2: Manifold) -> Tuple[torch.Tensor, torch.Tensor]:
-        # Need the log_map derivative which is not yet implemented
-        raise NotImplementedError
+        if jacobians is not None:
+            LieGroup._check_jacobians_list(jacobians)
+            dlog: List[torch.Tensor] = []
+            ret = diff.log_map(dlog)
+            jacobians.append(-diff.inverse().adjoint() @ dlog[0])
```
Comment (fantaosha) [inline theseus/geometry/lie_group.py:180]: The output shape of `SO2.adjoint()` is `(N,1,1)`, whereas that of `SO2.log_map()` is `(N,1)`, which might raise errors here. @luisenp

File: theseus/geometry/so2.py [hunk 2]
```diff
@@ -146,6 +147,7 @@ def _log_map_impl(
                 torch.ones(
                     self.shape[0],
                     1,
+                    1,
```
Comment (fantaosha) [inline theseus/geometry/so2.py:150]: Added this line so that the output shape is `(N,1,1)`.

**Review by mhmukadam — APPROVED** [review]
Looks good to merge if the discussion around the dim of logmap / dlog is resolved.

**Review by luisenp — APPROVED** [review]
LGTM! Feel free to merge.
